src/wizards/tokenSwapDealWizard/stages/termsStage/termsStage.ts

```diff
@@ -13,6 +14,7 @@ export class TermsStage implements IBaseWizardStage {
 
   termClauses: TermClause[] = [];
   hasUnsavedChanges = false;
+  stageMetadata: Partial<{ termsViewModes: EditingCard["viewMode"][] }> = {};
```

Comment:
```suggestion
  stageMetadata: { termsViewModes?: EditingCard["viewMode"][] } = {};
```

Reply:
Played around a bit.
Following makes the `this.stageSettings` type a bit clearer imo

![grafik](https://user-images.githubusercontent.com/30693990/157549768-6e9a5861-3816-4234-9dc6-bce95e261480.png)
```ts
type StageSettings = WizardManager["additionalStageMetadata"][number];
type TermsStageSettings = { termsViewModes?: EditingCard["viewMode"][] } & StageSettings

@autoinject
export class TermsStage implements IBaseWizardStage {
  public wizardManager: any;
  public wizardState: IWizardState<IDealRegistrationTokenSwap>;

  termClauses: TermClause[] = [];
  hasUnsavedChanges = false;
  stageSettings: TermsStageSettings = {};

  constructor(
    public wizardService: WizardService,
  ) {
  }

  activate(stageMeta: IStageMeta<StageSettings>): void {
```

Reply:
mhm. Yes. Maybe we should add the `StageSettings` type in some place because it will be used many times.

But, there is a catch: `TermsStageSettings`  with `StageSettings` gives access to all the stage's setting and not only to what the users defines in the stage. This means they can access everything and they also need to define the types of those.

I would go with just `type TermsStageSettings = { termsViewModes?: EditingCard["viewMode"][] }` for now until we get to a point where we need to access more than the stage supposed to access